pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -64,34 +64,53 @@ const (
 )
 
 var (
-	deploymentTemplatesMap map[string]*template.Template
+	serviceTemplate      *template.Template
+	deploymentTemplate   *template.Template
+	vmDeploymentTemplate *template.Template
 )
 
-func prepareTemplate(file fs.FileInfo, templateDir string) *template.Template {
-	if yaml, err := os.ReadFile(path.Join(templateDir, file.Name())); err != nil {
-		panic(fmt.Sprintf("unable to read template file %s : %v", file.Name(), err))
+func init() {
+	serviceYAMLPath := path.Join(env.IstioSrc, "pkg/test/framework/components/echo/kube/templates", echo.ServiceTemplateFile)
+	if filepath.IsAbs(echo.ServiceTemplateFile) {
+		serviceYAMLPath = echo.ServiceTemplateFile
+	}
+
+	if serviceYAML, err := os.ReadFile(serviceYAMLPath); err != nil {
+		panic(fmt.Sprintf("unable to read template file %s : %v", serviceYAMLPath, err))
 	} else {
-		yamlTemplate := template.New(file.Name())
-		if _, err1 := yamlTemplate.Funcs(sprig.TxtFuncMap()).Parse(string(yaml)); err1 != nil {
-			panic(fmt.Sprintf("unable to parse template %s : %v", file.Name(), err1))
+		serviceTemplate = template.New("echo_service")
```

Comment:
Rather than do all of this, let's add a method to https://github.com/istio/istio/blob/master/pkg/test/util/tmpl/parse.go:

```golang
func MustParse(tpl string) *template.Template {
	t.Helper()
	tpl2, err := Parse(tpl)
	if err != nil {
		panic(fmt.Sprintf("tmpl.ParseOrFail: %v", err))
	}
	return tpl2
}
```

Then all of this could be reduced to:

```golang
serviceTemplate := tmpl.MustParse(file.MustAsString("..."))
deploymentTemplate := tmpl.MustParse(file.MustAsString("..."))
vmDeploymentTemplate := tmpl.MustParse(file.MustAsString("..."))
```